index.js

```diff
@@ -3,17 +3,28 @@
 var fs = require('fs')
 var path = require('path')
 var resolve = require('resolve-from').silent
-var npmPrefix = require('npm-prefix')()
+var readNpmConfig = require('libnpmconfig').read
 
 module.exports = loadPlugin
 loadPlugin.resolve = resolvePlugin
 
+var windows = process.platform === 'win32'
+/* istanbul ignore next */
+var builtinNpmConfig =
+  windows && process.env.APPDATA
+    ? {
+        // C:\Users\{username}\AppData\Roaming\npm
+        prefix: path.join(process.env.APPDATA, 'npm')
+      }
+    : null
```

Comment:
I’m not opposed to using `libnpmconfig`, but this is a different change, that you didn’t mention. Care to elaborate?

Reply:
I'm not sure how to describe things step by step. It's mostly intuition. But I'll try.

https://docs.npmjs.com/files/npmrc.html describes four npmrc file locations:
- per-project config file (/path/to/my/project/.npmrc)
- per-user config file (~/.npmrc)
- global config file ($PREFIX/etc/npmrc)
- npm builtin config file (/path/to/npm/npmrc)
Those are not crossplatform locations.

https://docs.npmjs.com/misc/config has:
- per-project configuration file (/path/to/my/project/.npmrc)
- per-user configuration file (defaults to $HOME/.npmrc; configurable via CLI option --userconfig or environment variable $NPM_CONFIG_USERCONFIG)
- global configuration file (defaults to $PREFIX/etc/npmrc; configurable via CLI option --globalconfig or environment variable $NPM_CONFIG_GLOBALCONFIG)
- npm’s built-in configuration file (/path/to/npm/npmrc)
Here we have more crossplatform description, and also mentions of environment variables by which locations can be changed.

https://docs.npmjs.com/files/folders.html has description of where `npm prefix` locates.
>> The prefix config defaults to the location where node is installed. On most systems, this is /usr/local. On Windows, it’s %AppData%\npm.
It sounds a bit contradictory for windows though, because 'the location where node is installed' != `%AppData%\npm`.

What 'libnpmconfig' does? It solves for us resolution for:
- per-project configuration file (/path/to/my/project/.npmrc)
- per-user configuration file (defaults to $HOME/.npmrc; configurable via CLI option --userconfig or environment variable $NPM_CONFIG_USERCONFIG)
- global configuration file (defaults to $PREFIX/etc/npmrc; configurable via CLI option --globalconfig or environment variable $NPM_CONFIG_GLOBALCONFIG)
except for - npm’s built-in configuration file (/path/to/npm/npmrc)

Where npm-cli itself reads npm's built-in configuration file?
I have found the answer in this sequence:
- `{npm-cli-package}/bin/npm-cli.js` calls `npm.load(...)`
- `npm.load` is in `{npm-cli-package}/lib/npm.js` and contains `var builtin = path.resolve(__dirname, '..', 'npmrc')` and `npmconf.load(cli, builtin,...`
So, for the very `npm-cli` builtin configuration is `path.resolve(__dirname, '..', 'npmrc')` which on windows is `C:\Program Files\nodejs\node_modules\npm\npmrc` which contains `prefix=${APPDATA}\npm` which is `prefix: path.join(process.env.APPDATA, 'npm')` in this pull request.

So `prefix: path.join(process.env.APPDATA, 'npm')` is the missed piece in `libnpmconfig` for windows.

Have I missed something?

Reply:
Wonderful!

Reply:
Could you add a (shorter 😉) comment about `APPDATA` on Windows?